lib/c-api/build.rs

```diff
@@ -389,3 +390,44 @@ fn exclude_items_from_wasm_c_api(builder: Builder) -> Builder {
         .exclude_item("wasmer_engine_t")
         .exclude_item("wat2wasm")
 }
+
+fn build_inline_c_env_vars() {
+    use std::ffi::OsStr;
+
+    let mut shared_object_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+
+    assert_eq!(shared_object_dir.file_name(), Some(OsStr::new("c-api")));
+    shared_object_dir.pop();
+
+    assert_eq!(shared_object_dir.file_name(), Some(OsStr::new("lib")));
+    shared_object_dir.pop();
+
+    shared_object_dir.push("target");
+    shared_object_dir.push(env::var("PROFILE").unwrap());
+
+    let shared_object_dir = shared_object_dir.as_path().to_string_lossy();
+    let include_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+
+    // The following options mean:
+    //
+    // * `-I`, add `include_dir` to include search path,
+    // * `-L`, add `shared_object_dir` to library search path,
+    // * `-D_DEBUG`, enable debug mode to enable `assert.h`.
+    println!(
+        "cargo:rustc-env=INLINE_C_RS_CFLAGS=-I{I} -L{L} -D_DEBUG",
```

Comment:
(No action required.) I think I see why this didn't work: `-L` isn't really a compiler flag, it's a linker flag. It'll be ignored when not linking, and since it's not passed along to the linker (not in `INLINE_C_RS_LDFLAGS`) it would cause `-lwasmer_c_api` to fail because the linker wouldn't know to search our shared_object_dir.